core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -0,0 +1,175 @@
+package org.wildfly.extras.creaper.core.online;
+
+
+import org.apache.http.HttpHost;
+import org.apache.http.HttpStatus;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.MalformedChallengeException;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.AuthCache;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.auth.DigestScheme;
+import org.apache.http.impl.client.BasicAuthCache;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.OperationMessageHandler;
+import org.jboss.as.controller.client.OperationResponse;
+import org.jboss.dmr.ModelNode;
+import org.jboss.threads.AsyncFuture;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Controller provides execution of {@link ModelNode} or {@link Operation} over HTTP.
+ * Asynchronous execution is not provided. Controller does not inform about any progress, therefore, there is no point
+ * in using {@link OperationMessageHandler}.
+ * If an error occurs (server does not respond 401 or header does not contain WWW-Authenticate field after
+ * first request) {@link IllegalStateException} is thrown.
+ * Execution on {@link Operation} is allowed with <b>no attachments</b>.
+ */
+class HttpModelControllerClient implements ModelControllerClient {
+    private static final int NO_TIMEOUT = 0;
+
+    private final CredentialsProvider credsProvider;
+    private final HttpHost target;
+    private final String url;
+    private final RequestConfig requestConfig;
+    private CloseableHttpClient httpClient;
+    private HttpClientContext localContext;
+
+    public HttpModelControllerClient(String host, int port, String username, String password) {
+        this(host, port, username, password, NO_TIMEOUT);
+    }
+
+    public HttpModelControllerClient(String host, int port, String username, String password, int timeoutMillis) {
+        target = new HttpHost(host, port, "http");
+        credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(
+                new AuthScope(target.getHostName(), target.getPort()),
+                new UsernamePasswordCredentials(username, password));
+
+        RequestConfig.Builder configBuilder = RequestConfig.copy(RequestConfig.DEFAULT);
+        if (timeoutMillis != NO_TIMEOUT) {
+            configBuilder.setConnectTimeout(timeoutMillis).setSocketTimeout(timeoutMillis);
```

Comment:
Both of them are fine. I'd probably go for the first one, but if the variable name was too short, I'd probably switch to the second. I'll leave the decision up to you.